Fix yq usage in make verify

**What steps did you take and what happened:**
We added yq into the Makefile in order to properly process yaml templates after they've been processed by newer versions of kustomize (4.5.2+). Unfortunately, something is wrong and yq is not being found by the make process.
```sh
bash: line 1: /home/runner/work/cluster-api-provider-packet/cluster-api-provider-packet/hack/tools/bin/yq-v4.35.2: No such file or directory
make[1]: *** [Makefile:295: generate-templates] Error 127
make[1]: Leaving directory '/home/runner/work/cluster-api-provider-packet/cluster-api-provider-packet'
make: *** [Makefile:285: generate] Error 2
```
**What did you expect to happen:**
make verify to work

--- a/Makefile
+++ b/Makefile
@@ -285,7 +285,7 @@ generate: ## Generate code
 	$(MAKE) generate-templates
 
 .PHONY: generate-templates
-generate-templates: $(KUSTOMIZE) ## Generate cluster templates
+generate-templates: $(KUSTOMIZE) $(YQ) ## Generate cluster templates
 	$(KUSTOMIZE) build templates/development --load-restrictor LoadRestrictionsNone > templates/cluster-template-development.yaml
 	$(KUSTOMIZE) build templates/experimental-emlb --load-restrictor LoadRestrictionsNone > templates/cluster-template-emlb.yaml
 	$(KUSTOMIZE) build templates/experimental-emlb-crs-cni --load-restrictor LoadRestrictionsNone > templates/cluster-template-emlb-crs-cni.yaml

